### PR TITLE
Clean up virt-handler to make it testable

### PR DESCRIPTION
**Pull request by rmohr**
Clean up virt-handler to make it testable

### CODE REVIEW

File: cmd/virt-handler/virt-handler.go [hunk 1]
```diff
@@ -136,22 +138,25 @@ func main() {
...
 			err = domain.Destroy()
 			if err != nil {
 				fmt.Println(err)
+				// If the VM is already destroyed, we still want to try to undefine the VM
 				if code := err.(libvirtapi.VirError).Code; code != libvirtapi.VIR_ERR_OPERATION_INVALID {
 					return cache.ErrRequeue{Err: err}
 				}
 			}
-			domain.Undefine()
+			err = domain.Undefine()
```
Comment (stu-gott) [inline cmd/virt-handler/virt-handler.go]: oops! good catch!

**Comment by stu-gott** [comment]
Looks good. In general, what condition were the 1-second sleeps addressing?

**Comment by rmohr** [comment]
> Looks good. In general, what condition were the 1-second sleeps addressing?

They are there to avoid hot-looping. For now we block the whole worker for one second. We could now implement our own BackOff queue, but kubernetes already has the workerqueue.Interface and it hopefully lands in client-go soon.

**Comment by fabiand** [comment]
My bad - I still need to get used to that all patches are show squashed in the "files changed" view.

**Pull request merged**